### PR TITLE
REF: back DatetimeBlock, TimedeltaBlock by DTA/TDA

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

Broken off from #40149.   Like #40149, this backs DTBlock/TDBlock by (sometimes 2D) DTA/TDA.  Unlike #40149, this does _not_ make the DTA backing DatetimeTZBlock sometimes-2D.

### CODE REVIEW

File: pandas/core/sorting.py [hunk 1]
```diff
@@ -417,11 +418,16 @@ def nargminmax(values, method: str):
     mask = np.asarray(isna(values))
     values = values._values_for_argsort()
 
-    idx = np.arange(len(values))
+    idx = np.arange(values.shape[axis])
+    if values.ndim > 1:
+        if mask.any():
```
Comment (jreback) [inline pandas/core/sorting.py:423]: is this hit in tests?

**Review by jreback — CHANGES_REQUESTED** [review]
also pls rebase

File: pandas/core/sorting.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/sorting.py:423]: the case with mask.any() being True?  no

**Comment by jbrockmendel** [comment]
rebased + green.  between this and #40477 we're within striking distance of ripping out a couple more Block subclasses

File: pandas/core/sorting.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/sorting.py:421]: updated to implement this, with tests that hit it

**Review by jreback — CHANGES_REQUESTED** [review]
this looks pretty good, just a small comment

File: pandas/core/array_algos/take.py [hunk 2]
```diff
@@ -92,7 +94,12 @@ def take_nd(
 
     if not isinstance(arr, np.ndarray):
         # i.e. ExtensionArray,
-        # includes for EA to catch DatetimeArray, TimedeltaArray
+        if arr.ndim == 2:
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/array_algos/take.py:97]: I considered that, but decided this is a placeholder until the more specific check from #40149 takes its place

File: pandas/core/frame.py [hunk 3]
```diff
@@ -9541,6 +9541,9 @@ def func(values: np.ndarray):
 
         def blk_func(values, axis=1):
             if isinstance(values, ExtensionArray):
+                if values.ndim == 2:
```
Comment (jreback) [inline pandas/core/frame.py:9544]: isn't this just `axis=axis` ?

**Review by jreback — COMMENTED** [review]
just a question. what's failing on master?

File: pandas/core/frame.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/core/frame.py:9544]: sure.  next pass

**Comment by jbrockmendel** [comment]
> just a question. what's failing on master?

havent looked this morning, but last night it was something td64/dt64-casting related merged into npdev

File: pandas/core/internals/blocks.py [hunk 4]
```diff
@@ -1172,6 +1174,7 @@ def _interpolate_with_fill(
             limit_area=limit_area,
         )
 
+        values = maybe_coerce_values(values)
```
Comment (jreback) [inline pandas/core/internals/blocks.py:1177]: prob can have maybe_coerce_values do downcasting (maybe)

**Review by jreback — APPROVED** [review]

**Pull request merged**